d2ir/compile.go

```diff
@@ -167,6 +188,12 @@ func (c *compiler) resolveSubstitutions(varsStack []*Map, node Node) {
 						c.errorf(node.LastRef().AST(), `cannot substitute map variable "%s" as part of a string`, strings.Join(box.Substitution.IDA(), "."))
```

Comment:
needs to be updated.

Reply:
Also should check for composite here as both could be nil.

Reply:
both could be nil is handled in the compiler (cleaner I felt). there's a test for it. `bad-var`

Reply:
I don't see how that's cleaner nor why it's in `d2compiler`. The IR compiler should be complaining about that. Variables are not the responsibility of `d2compiler`.

Reply:
nvm it was wrong behavior anyway. no composite and no primary should be allowed (e.g. maybe people want to spread), but it's a mistake when the resolved variable itself has no composite and no primary, which is right here.